src/Weft/Generics/QueryParser.hs

```diff
@@ -34,44 +42,62 @@ type Vars = M.Map String String
 ------------------------------------------------------------------------------
 -- |
 class GPermFieldsParser (rq :: * -> *) where
-  gPermFieldsParser :: Permutation (ReaderT Vars Parser) (rq x)
+  gPermFieldsParser :: [ReaderT Vars Parser (rq x)]
 
 instance {-# OVERLAPPABLE #-} GPermFieldsParser fq => GPermFieldsParser (M1 a b fq) where
-  gPermFieldsParser = M1 <$> gPermFieldsParser
+  gPermFieldsParser = fmap M1 <$> gPermFieldsParser
 
 instance ( GPermFieldsParser fq
          , GPermFieldsParser gq
+         , forall x. (Monoid (fq x), Monoid (gq x))
          ) => GPermFieldsParser (fq :*: gq) where
-  gPermFieldsParser = (:*:) <$> gPermFieldsParser
-                            <*> gPermFieldsParser
+  gPermFieldsParser =
+      (fmap (:*: mempty) <$> gPermFieldsParser @fq)
```

Comment:
Have to say I don't quite understand why you are fmapping a product with mempty over gPermFieldsParser rather than applying the product like before. This isn't a critique, just my own confusion!

Reply:
good thing we went over it together!